### PR TITLE
Add observability instrumentation for Prometheus receiver

### PR DESCRIPTION
**Pull request by dinooliva**
The pr instrument the Prometheus receiver using the observability metrics library. Adding this instrumentation will allow users to monitor service that rely on this receiver.

### CODE REVIEW

**Comment by dinooliva** [comment]
@bogdandrutu @songy23 - ptal

File: receiver/prometheusreceiver/metrics_receiver.go [hunk 1]
```diff
@@ -138,6 +139,7 @@ func (pr *Preceiver) StartMetricsReception(host receiver.Host) error {
 		ctx := host.Context()
 		c, cancel := context.WithCancel(ctx)
 		pr.cancel = cancel
+		c = observability.ContextWithReceiverName(c, pr.MetricsSource())
```
Comment (bogdandrutu) [inline receiver/prometheusreceiver/metrics_receiver.go]: For the tag value you should probably use the observability.ComponentName that includes also the type + name from the component config.

Comment (dinooliva) [reply receiver/prometheusreceiver/metrics_receiver.go]: @bogdandrutu - done, ptal

File: receiver/prometheusreceiver/metrics_receiver.go [hunk 2]
```diff
@@ -138,6 +139,7 @@ func (pr *Preceiver) StartMetricsReception(host receiver.Host) error {
 		ctx := host.Context()
 		c, cancel := context.WithCancel(ctx)
 		pr.cancel = cancel
+		c = observability.ContextWithReceiverName(c, observability.MakeComponentName(typeStr, pr.MetricsSource()))
```
Comment (bogdandrutu) [inline receiver/prometheusreceiver/metrics_receiver.go]: This should be config.Type() and Config.Name() (the whole config not the Prometheus special config)

**Pull request merged**